### PR TITLE
server.prpc: RegisterDefaultAuth accepts a nil authenticator now

### PR DESCRIPTION
**Pull request by azr**

### CODE REVIEW

File: server/prpc/server.go [hunk 1]
```diff
@@ -111,8 +111,7 @@ func (s *Server) authenticate() router.Middleware {
 	if a == nil {
 		a = GetDefaultAuth()
 		if a == nil {
-			panic("prpc: no custom Authenticator was provided and default authenticator was not registered. " +
```
Comment (danjacques) [inline server/prpc/server.go]: I'd actually prefer we leave the panic in. The rationale here is that if someone explicitly wants no auth, they can install an empty default auth (`[]auth.Authenticator(nil)`), but if they forget to install auth, that's a problem worth panicking over.

Comment (azr) [inline server/prpc/server.go]: I totally understand your point, but in go `[]auth.Authenticator(nil) == nil`

Comment (danjacques) [inline server/prpc/server.go]: Yeah sorry, I should have said, `[]auth.Authenticator{}`.

Perhaps the better option here is to declare a package global called `NoAuthenticator == []auth.Authenticator{}` and have the user supply that if they want no authenticator.

Comment (azr) [inline server/prpc/server.go]: ah ! cool, I'll add that right away :)

File: server/prpc/server.go [hunk 2]
```diff
@@ -37,6 +37,10 @@ var (
 	// exposeHeaders lists the whitelisted non-standard response headers that the
 	// client may accept.
 	exposeHeaders = strings.Join([]string{prpccommon.HeaderGRPCCode}, ", ")
+
+	//NoAuthenticator should be used with a Server to explicitly specify it wont use any Auth
```
Comment (danjacques) [inline server/prpc/server.go]: nits: spaces after "//", period after "panic", missing apostrophe in "won't", "Auth" isn't a keyword (did you mean, "Authenticator"), and please reword to mention the `RegisterDefaultAuth` function name, something like:
NoAuthenticator can be used with RegisterDefaultAuth or Server.Authenticator to explicitly specify that your Server won't apply any authentication to incoming connections.

File: server/prpc/server.go [hunk 3]
```diff
@@ -37,6 +37,10 @@ var (
 	// exposeHeaders lists the whitelisted non-standard response headers that the
 	// client may accept.
 	exposeHeaders = strings.Join([]string{prpccommon.HeaderGRPCCode}, ", ")
+
+	// NoAuthenticator can be used with RegisterDefaultAuth or Server.Authenticator to explicitly specify that
+	// your Server won't apply any authentication to incoming connections.
+	NoAuthenticator = auth.Authenticator{}
```
Comment (azr) [inline server/prpc/server.go]: there ! :)

**Comment by danjacques** [comment]
Looks good! On this project, we actually commit through the Chromium commit queue system (runs test suites against lots of hardware) rather than merging directly. Could you please post this according to [these instructions](https://github.com/luci/luci-go#contributing) and tag me as a reviewer?

Thanks for contributing!

**Comment by azr** [comment]
its up !  https://codereview.chromium.org/2151303002

**Pull request closed without merging**